### PR TITLE
Integrate with kepler action

### PR DESCRIPTION
**Pull request by SamYuan1990**
try to integration with latest kepler-action with features below to provide a tekton based k8s cluster for kepler's CI for model server training and validation.
1. set up container run time as docker.
2. use kepler action to install header and libbpf
3. use kepler action to set k8s cluster as kind for now
4. set up prometheus
5. set up tekton

todo in other PRs:
- for redhat support will be refined in other PR, as local dev cluster and kepler action need to be refined.
- for docker and other basic tooling.... keep them for now, or maybe we should considering a self build image on aws?
- for other container runtime backend and podman... in the future
- some refine works in local dev cluster.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,12 +47,13 @@ jobs:
 
   unitpreOS:
     strategy:
+      fail-fast: false
       matrix:
       # us-east-2
       # Ubuntu 20.04 AMI is ami-0e4d0bb9670ea8db0
       # ubuntu 22.04 AMI is ami-05fb0b8c1424f266b 
       # RHEL 9       AMI is ami-078cbc4c2d057c244
-        AMI_ID: [ami-0e4d0bb9670ea8db0,ami-05fb0b8c1424f266b,ami-078cbc4c2d057c244]
```
Comment (SamYuan1990) [inline .github/workflows/ci.yml:55]: @rootfs , it seems we have 2 rhel AMI id in the list, could you please check?

Comment (rootfs) [reply .github/workflows/ci.yml:55]: these are 2 ubuntu and 1 rhel

Comment (SamYuan1990) [reply .github/workflows/ci.yml:55]: but from the test output,
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516556954?pr=18#step:3:12
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516559265?pr=18#step:3:11

it seems we have two redhat instance?

Comment (SamYuan1990) [reply .github/workflows/ci.yml:55]: Machine name: 'ip-172-31-7-40'
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516556954?pr=18#step:3:12
Machine name: 'ip-172-31-7-40'
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516559265?pr=18#step:3:11
it seems the two jobs been set into same instance.

File: .github/workflows/ci_integration.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    needs: setup-runner
+    runs-on: [self-hosted, linux, x64]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Docker
+        id: docker
+        run: |
+            # Add Docker's official GPG key:
```
Comment (rootfs) [inline .github/workflows/ci_integration.yml:69]: add a test here to see which OS the script is running on. if it is RHEL, use yum.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SamYuan1990) [reply .github/workflows/ci.yml:55]: btw, @sunya-ch , may I know which ami id and instance configuration is used for model server for now?

Comment (sunya-ch) [reply .github/workflows/ci.yml:55]: Please find the configuration below:
```
kepler_model_server_image: quay.io/sustainable_computing_io/kepler_model_server:v0.7
ami_id: ami-0e4d0bb9670ea8db0
region: us-east-2
secuirty_group: sg-cda39bb6
volume_size: 120
key_name: kepler-ci-key
instance_list:
- m5zn.metal 
- i3.metal 
- m5dn.metal
- m5n.metal
- c5d.metal 
- c7i.metal-24xl
- i3en.metal 
- m7i.metal-24xl 
```

reference: https://github.com/sustainable-computing-io/kepler-model-training-playbook/blob/main/group_vars/local.yml

File: .github/workflows/ci_integration.yml (hunk 2 above)
Comment (SamYuan1990) [reply .github/workflows/ci_integration.yml:69]: as current our action with lots for apt command for ubuntu, I suppose for kepler action supports RHEL needs another topic, also for podman or runc...

**Comment by SamYuan1990** [comment]
```
./main.sh: line 47: HOME: unbound variable
fail to setup local-dev-cluster
```

**Comment by SamYuan1990** [comment]
query GHA for knowledge at https://github.com/orgs/community/discussions/112231

**Comment by sunya-ch** [comment]
The error seems to come from [local-dev-cluster](https://github.com/sustainable-computing-io/local-dev-cluster/tree/main).

https://github.com/sustainable-computing-io/local-dev-cluster/blob/11a3e4cbc5584b0fed4eb6aa8ce60103db3b0e58/main.sh#L47

It seems to happen in ec2 instance for multiple cases. for example,
- https://forum.buildkite.community/t/autoscaling-group-kept-creating-instances-in-loop/2924/13

How's about replacing that environment with full path.

**Comment by SamYuan1990** [comment]
it seems t2.micro is too small to run the kind cluster. https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8314627605/job/22752095043?pr=18#step:5:685

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SamYuan1990) [reply .github/workflows/ci.yml:55]: it seems t2.micro is too small to run the kind cluster. https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8314627605/job/22752095043?pr=18#step:5:685

I don't want CI charges a lot.  @rootfs , @sunya-ch , any suggestion for a minimal size instance?

Comment (sunya-ch) [reply .github/workflows/ci.yml:55]: You can check the spot price that @rootfs shared here: https://gist.github.com/rootfs/49855d0c75ef8a27e494eda6eb928aab

Also the spot instance I have tested with data collection here: https://github.com/sustainable-computing-io/kepler-model-training-playbook/blob/main/README.md

The cheapest one that can run seem to be `m5zn.metal`.

**Review by rootfs — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sunya-ch) [reply .github/workflows/ci.yml:55]: @SamYuan1990  Agree. I don't think we need all profile for CI testing. Also, we need to limit the path to activate this CI only when necessary. The list above in playbook repo is for collecting the data for power model training. We need to cover as much as possible. But, we do collect the data only once if there is no impacting change on Kepler exporter such as metric name.

**Pull request merged**